criu/kerndat.c

```diff
@@ -531,6 +571,31 @@ static bool kerndat_has_memfd_hugetlb(void)
 	return 0;
 }
 
+static bool kerndat_has_memfd_secret(void)
+{
+#if defined(CONFIG_HAS_MEMFD_SECRET)
+	int ret;
+
+	ret = memfd_secret(0);
+
+	if (ret > 0) {
+		kdat.has_memfd_secret = true;
+		close(ret);
+	} else if (ret == -1 && (errno == ENOSYS || errno == EINVAL || errno == EMFILE || errno == ENOMEM)) {
```

Comment:
`ENOSYS` means that `memfd_secret` system call is not available. All the rest `errno` values mean that there was an actual error.

Reply:
understood, thanks!